include/measurement_kit/nettest.hpp

```diff
@@ -0,0 +1,1922 @@
+    static constexpr const char *key = "status.measurement_submission";
+    
+    /// The "idx" attribute of this event.
+    int64_t idx = {};
+};
+
+/// C++ representation of the "status.measurement_done" event.
+class StatusMeasurementDoneEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.measurement_done";
+    
+    /// The "idx" attribute of this event.
+    int64_t idx = {};
+};
+
+/// C++ representation of the "status.report_close" event.
+class StatusReportCloseEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.report_close";
+    
+    /// The "report_id" attribute of this event.
+    std::string report_id = {};
+};
+
+/// C++ representation of the "status.report_create" event.
+class StatusReportCreateEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.report_create";
+    
+    /// The "report_id" attribute of this event.
+    std::string report_id = {};
+};
+
+/// C++ representation of the "status.resolver_lookup" event.
+class StatusResolverLookupEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.resolver_lookup";
+    
+    /// The "ip_address" attribute of this event.
+    std::string ip_address = {};
+};
+
+/// C++ representation of the "status.started" event.
+class StatusStartedEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.started";
+    /* No attributes */
+};
+
+/// C++ representation of the "status.update.performance" event.
+class StatusUpdatePerformanceEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.update.performance";
+    
+    /// The "direction" attribute of this event.
+    std::string direction = {};
+    
+    /// The "elapsed" attribute of this event.
+    double elapsed = {};
+    
+    /// The "num_streams" attribute of this event.
+    int64_t num_streams = {};
+    
+    /// The "speed_kbps" attribute of this event.
+    double speed_kbps = {};
+};
+
+/// C++ representation of the "status.update.websites" event.
+class StatusUpdateWebsitesEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.update.websites";
+    
+    /// The "url" attribute of this event.
+    std::string url = {};
+    
+    /// The "status" attribute of this event.
+    std::string status = {};
+};
+
+/// C++ representation of the "task_terminated" event.
+class TaskTerminatedEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "task_terminated";
+    /* No attributes */
+};
+
+} // namespace events
+
+/// Contains settings classes.
+namespace settings {
+
+/// Settings common to all nettests.
+class CommonSettings {
```

Comment:
In the FFI API there are two layers of options. The `settings` and inside of them the `options`. This actually happens for historical reasons. That is, because of how stuff is implemented internally. Logically, the distinction between settings and options does not make much sense. In general both contain stuff that you may not even want to configure unless you have specific needs. For this reason here all the common options and settings are part of the same class rather than nesting options inside settings.